src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -34,7 +36,7 @@ internal class TaskHost :
 #if FEATURE_APPDOMAIN
         MarshalByRefObject,
 #endif
-        IBuildEngine7
+        IBuildEngine7, IRarBuildEngine
```

Comment:
Two comments:
First, rather than making an IRarBuildEngine, it would be easier to make an IBuildEngine8 and implement that. That's particularly good because then it forces you to consider it in all places that currently use IBuildEngine7, since that will be essentially equivalent to all the things that might need RAR.

On the other hand, we generally want to have exhausted other options before doing something like this. Can you explain why you want to make an IRarBuildEngine?

Reply:
Related: https://github.com/dotnet/msbuild/pull/5613/files#r477380003. Will continue with the comment there.